packages/utils/src/ts-eslint/Rule.ts

```diff
@@ -632,33 +632,48 @@ interface RuleModule<
    */
   create(context: Readonly<RuleContext<TMessageIds, TOptions>>): TRuleListener;
 }
-type AnyRuleModule = RuleModule<string, readonly unknown[]>;
+export type AnyRuleModule = RuleModule<string, readonly unknown[]>;
 
-type RuleCreateFunction<
+/**
+ * A loose definition of the RuleModule type for use with configs. This type
+ * intended to relax validation of types so that we can have basic validation
+ * without being overly strict about nitty gritty details matching.
+ *
+ * For example the plugin might be declared using an old version of our types or
+ * they might use the DefinitelyTyped eslint types. Ultimately we don't need
+ * super strict validation in a config - a loose shape match is "good enough" to
+ * help validate the config is correct.
+ *
+ * See also LooseParserModule, LooseProcessorModule
```

Comment:
[Docs] Nit: for nice JSDoc linkage

```suggestion
 * @see {@link LooseParserModule}, {@link LooseProcessorModule}
```